### DISCUSSION TITLE
State of app development for elementary

### DISCUSSION THREAD

**Discussion by pongloongyeat** [elementary/streams#39] [Ideas]
Not sure if this is a stream idea but it could certainly serve as discussion for livestreams!

Currently, app development is pretty simple for new users to elementary, i.e. write some code in Code, meson build and voila but for people who are looking for better tooling (i.e. autocompletion with [Vala Language Server](https://github.com/benwaffle/vala-language-server), offline documentation, performance profiling, etc.) this is quite lacking.

GNOME has Builder, KDE has KDevelop. Would we see elementary have its own IDE in the future? Builder is probably sufficient in this regard but I'd still much rather have elementary's opinion on this matter :)

**Comment by danirabbit** [comment]
Yeah this could be an interesting UX study for @cassidyjames maybe: setting up a development environment from scratch and identifying pain points and complexity

I know language server support in Code is something we would like to have and there are a couple of Vala language servers being developed by 3rd parties. We just need someone to take on the work
